### PR TITLE
Try an alternate approach for non-latin fonts

### PR DESCRIPTION
**Pull request by kjellr**
This PR is a slightly different version of the method proposed by @mako09 in #633. It also uses the `:lang()` pseudo class alongside a Sass `@each` loop to swap out fonts, but with a couple differences: 

-  Rather than repeat a big code block each time, this PR replaces all `font-family` rules with a [`font-family()` mixin](https://github.com/WordPress/twentynineteen/blob/746b775633762a112ab8c3230889d92170b300eb/sass/mixins/_mixins-master.scss#L117-L126) that includes non-latin language support automatically. This keeps our sass clean.  
- Instead of offering both serif and sans-serif fonts, this keeps things simple by sticking with just one font-family. It uses the same [font family fallbacks](https://github.com/WordPress/twentynineteen/blob/746b775633762a112ab8c3230889d92170b300eb/sass/variables-site/_fonts.scss#L19-L46) we have in place now, so if we implement this, end users should have no visible font difference on their site. 

These bits of code do most of the work in this PR: 

https://github.com/WordPress/twentynineteen/blob/746b775633762a112ab8c3230889d92170b300eb/sass/mixins/_mixins-master.scss#L117-L126

https://github.com/WordPress/twentynineteen/blob/746b775633762a112ab8c3230889d92170b300eb/sass/variables-site/_fonts.scss#L19-L46

The main benefit of this approach is that it avoids `!important` styles completely (fixing #719 and eliminating the hack in #685). It does add more CSS, but that seems like it's probably required if we're going to do this the right way.

**Known issues:**  

- [x] Some languages don't currently work in Chrome (Try `ar` for instance)
- [x] As noted [in the other thread](
https://github.com/WordPress/twentynineteen/pull/633#issuecomment-439712175), sans-serif menu text and headlines (`entry-title`, `h1`, `h2`, etc.) don't appear to inherit the correct font on the front end, despite the fact that the  CSS (appears to be?) correct.

### CODE REVIEW

**Comment by kjellr** [comment]
>  Some languages don't currently work in Chrome (Try ar for instance)
> As noted in the other thread, sans-serif menu text and headlines (entry-title, h1, h2, etc.) don't appear to inherit the correct font on the front end, despite the fact that the CSS (appears to be?) correct.

Both of these may just be issues with my testing actually... I'm seeing them why I manually adjust the `lang` html attribute on an already-loaded page, but not when I set the site language via WP-Admin.

**Comment by mako09** [comment]
> * Rather than repeat a big code block each time, this PR replaces all `font-family` rules with a [`font-family()` mixin](https://github.com/WordPress/twentynineteen/blob/746b775633762a112ab8c3230889d92170b300eb/sass/mixins/_mixins-master.scss#L117-L126) that includes non-latin language support automatically. This keeps our sass clean.

Great! I strongly agree with this approach.

> * Instead of offering both serif and sans-serif fonts, this keeps things simple by sticking with just one font-family. It uses the same [font family fallbacks](https://github.com/WordPress/twentynineteen/blob/746b775633762a112ab8c3230889d92170b300eb/sass/variables-site/_fonts.scss#L19-L46) we have in place now, so if we implement this, end users should have no visible font difference on their site.

I can't agree with this point.
One of the features of Twentynineteen is
> Featuring ample whitespace, and modern sans-serif headlines paired with classic serif body text, Twenty Nineteen is built to be beautiful on the go.

as you mentioned at https://wordpress.org/news/2018/12/bebo/#mce_18. Users in non-Latin languages ​​are excluded from the benefit from the beginning.

Two sets for body and headings should be prepared separately. Under that condition, it should be the users to decide the fonts to be set for each even if there are the same.

File: style.css [hunk 1]
```diff
@@ -462,6 +619,396 @@ h6 {
...
+.error-404 .page-title:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
+.post-navigation .post-title:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
+.page-links:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
+.page-description:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
+.pagination .nav-links:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
+.sticky-post:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
+.site-title:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
+.site-info:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
+#cancel-comment-reply-link:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
+img:after:lang(ar, ary, azb, ckb, fa-IR, haz, ps),
```
[the remark this answers is not in the document]
Comment (mako09) [reply style.css]: Is `img:after{font-family:..}` necessary in the first place?

**Comment by kjellr** [comment]
> Users in non-Latin languages ​​are excluded from the benefit from the beginning.

I definitely see the point here, but I'm also very weary of changing the fonts for non-latin users (with no way of changing back) after the theme has already been released. As a user, I'd find that to be very jarring.

File: style.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (kjellr) [reply style.css]: Good point. That rule isn't necessary. I've removed it in af182ac. 👍

**Comment by kjellr** [comment]
Pushed a small update in e17a47b to use a [Sass placeholder](https://sass-lang.com/guide#topic-7) to generate the non-latin CSS rules. This groups them all together in `style.css`, and saves ~25k. 

h/t @allancole for the idea.

**Comment by mako09** [comment]
I’m sorry that my explanation was’t clear enough.My idea is as follows.

Definitions in mixins/_mixins-master.scss are like
```
/* Ensure all font family declarations come with non-latin fallbacks */
@mixin body-font-family( $font_family: $font__body ) {
	font-family: $font_family;
	@extend %non-latin-body-fonts;
}

@mixin font-family( $font_family: $font__heading ) {
	font-family: $font_family;
	@extend %non-latin-fonts;
}

/* Build our non-latin font styles */
%non-latin-body-fonts {
	@each $lang, $font__fallback in $font__body__fallbacks {
		&:lang(#{$lang}) {
			font-family: unquote( $font__fallback );
		}
	}
}

%non-latin-fonts {
	@each $lang, $font__fallback in $font__fallbacks {
		&:lang(#{$lang}) {
			font-family: unquote( $font__fallback );
		}
	}
}
```
and in variables-site/_fonts.scss, 
```
$font__body__fallbacks: (
...
);
$font__fallbacks: (
...
);
```
where `$font__body__fallbacks` is exactly the same as `$font__fallbacks` **at this time**.
Size of style.css grows a little bit, but its work does not change at all. All current users are not confused.

**In future**, users who want to change the font for body from sans-serif to serif can easily do it.

**Comment by kjellr** [comment]
Hey @mako09 — Thanks for sharing your example. I understand the how we'd implement this, but I still feel that it doesn't make sense to do so at this stage, [for the reasons explained above](
https://github.com/WordPress/twentynineteen/pull/728#issuecomment-446586070). 

> where $font__body__fallbacks is exactly the same as $font__fallbacks at this time.
Size of style.css grows a little bit, but its work does not change at all. 

If we were adding unique fallbacks for both serif and sans-serif fonts right now, this approach would make sense. But since these two variables and mixins would essentially be the same, I don't think it's a good idea to leave repetitive markup in there.

**Comment by mako09** [comment]
In my understanding, the flow of discussion at #29 was as follows.
It was not well considered which of serif and sans-serif is suitable font for body of Twentynineteen.
Then, the fonts for the body were copied from Twentyseventeen which was originally designed with sans-serif.

Anyway, regardless of my thoughts, I can understand what you say.
This PR is very good, please proceed as it is. 

Thank you.

**Comment by allancole** [comment]
I also agree with Kjell’s point and wouldn’t want users to experience unexpected font changes. All things considered, I think this is ready to go as-is 👍

**Comment by kjellr** [comment]
This PR has been ported over to Trac: 

https://core.trac.wordpress.org/ticket/45731

**Comment by kjellr** [comment]
Since this has already been transferred over to Trac, I'm going to go ahead and close this PR for now. When this gets merged in, it'll happen over on Trac. 👍 

Continue following here for more updates: 
https://core.trac.wordpress.org/ticket/45731

Thanks everyone!

**Pull request closed without merging**